src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -130,54 +123,105 @@ int IVsEditorFactory.Close()
                 case "Form":
 
                     // We must create the WinForms designer here
-                    var loaderName = GetWinFormsLoaderName(vsHierarchy);
-                    var designerService = (IVSMDDesignerService)_oleServiceProvider.QueryService<SVSMDDesignerService>();
-                    var designerLoader = (IVSMDDesignerLoader)designerService.CreateDesignerLoader(loaderName);
-                    if (designerLoader is null)
-                    {
-                        goto case "Code";
-                    }
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetWinFormsDesignerData(vsHierarchy, itemid, textBuffer, readOnlyStatus, initializeView: true);
+                    break;
 
-                    try
-                    {
-                        designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, (IVsTextLines)textBuffer);
-                        pbstrEditorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
+                case "Code":
 
-                        var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
-                        ppunkDocView = Marshal.GetIUnknownForObject(designer.View);
-                        pguidCmdUI = designer.CommandGuid;
-                    }
-                    catch
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetCodeEditorData(textBuffer, readOnlyStatus, initializeView: true);
+                    break;
+
+                default:
+
+                    return VSConstants.E_INVALIDARG;
+            }
+
+            ppunkDocData = Marshal.GetIUnknownForObject(textBuffer);
+
+            return VSConstants.S_OK;
+        }
+
+        public object GetDocumentData(uint grfCreate, string pszMkDocument, IVsHierarchy pHier, uint itemid)
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
+
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            IVsTextBuffer? textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
+
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
+
+                    if (ErrorHandler.Failed(hresult))
                     {
-                        designerLoader.Dispose();
-                        throw;
+                        Marshal.ThrowExceptionForHR(hresult);
```

Comment:
`Marshal.ThrowExceptionForHR` is a no-op for non-failure HRESULTs, so you don't need the `ErrorHandler.Failed` check.

Reply:
Thanks, done.